plugin/pkg/auth/authorizer/node/node_authorizer.go

```diff
@@ -119,6 +122,30 @@ func (r *NodeAuthorizer) authorizeGet(nodeName string, startingType vertexType,
 	return ok, "", nil
 }
 
+// authorizeImpersonate authorizes "impersonate" requests to serviceaccounts of pods running on a nade
+func (r *NodeAuthorizer) authorizeImpersonate(nodeName string, startingType vertexType, attrs authorizer.Attributes) (bool, string, error) {
+	if attrs.GetVerb() != "impersonate" || len(attrs.GetName()) == 0 {
+		glog.V(2).Infof("NODE DENY: %s %#v", nodeName, attrs)
+		return false, "can only impersonate individual resources of this type", nil
```

Comment:
nit: error message isn't accurate if `attrs.GetVerb() != "impersonate"`

Reply:
looks like the other (preexisting) methods make the same mistake, if you feel like cleaning up.

Reply:
The "individual" makes the error message accurate.